### ISSUE TITLE
permission error with yarn global add

### ISSUE THREAD

**Issue by wangnengjie** [termux/termux-packages#4664]
**Problem description**
Every time I use `yarn global add xxx` will get a permisson denied error. I first think it is caused by root permisson. Since I did not root my phone, I use proot. But get another error. I am not really familiar with that. I wonder whether it is caused by Android 10.

**Steps to reproduce**
```
yarn global add serve \\get permission denied
unset LD_PRELOAD
termux-chroot \\another error I will paste below
```

**Additional information**
- yarn-error.log
```
Arguments:
  /data/data/com.termux/files/usr/bin/node /data/data/com.termux/files/usr/share/yarn/bin/yarn.js global add serve

PATH:
  /data/data/com.termux/files/home/bin:/usr/local/bin:/data/data/com.termux/files/home/.cargo/bin:/data/data/com.termux/files/usr/bin:/data/data/com.termux/files/usr/bin/applets:/data/data/com.termux/files/home/go/bin

Yarn version:
  1.19.2

Node version:
  13.0.0

Platform:
  android arm64

Trace:
  Error: EACCES: permission denied, copyfile '/data/data/com.termux/files/home/.cache/yarn/v6/npm-ansi-styles-3.2.1-41fbb20243e50b12be0f04b8dedbf07520ce841d-integrity/node_modules/ansi-styles/license' -> '/data/data/com.termux/files/home/.config/yarn/global/node_modules/ansi-styles/license'

npm manifest:
  No manifest

yarn manifest:
  No manifest

Lockfile:
  No lockfile
```
- termux-chroot error
```
proot error: execve("/usr/bin/zsh"): Function not implemented
proot info: possible causes:
  * the program is a script but its interpreter (eg. /bin/sh) was not found;
  * the program is an ELF but its interpreter (eg. ld-linux.so) was not found;
  * the program is a foreign binary but qemu was not specified;
  * qemu does not work correctly (if specified);
  * the loader was not found or doesn't work.
fatal error: see `proot --help`.
proot error: trying to remove a directory outside of '/data/data/com.termux/files/usr/tmp', please report this error.
proot error: can't chmod '/data/data/com.termux/files/usr/tmp/proot-6957-Cv8mja': No such file or directory
```
- termux-info
```
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.org/packages/ stable main
# root-repo (sources.list.d/root.list)
deb https://dl.bintray.com/grimler/termux-root-packages-24 root stable
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable
# unstable-repo (sources.list.d/unstable.list)
deb https://dl.bintray.com/xeffyr/unstable-packages unstable main
Updatable packages:
All packages up to date
Android version:
10
Kernel build information:
Linux localhost 4.14.117-perf+ #1 SMP PREEMPT Sat Nov 30 01:22:12 CST 2019 aarch64 Android
Device manufacturer:
OnePlus
Device model:
GM1910
```

**Comment by Grimler91** [comment]
Does `~/.config` exist? If not, try creating it and try `yarn global add serve again`

**Comment by wangnengjie** [comment]
> Does `~/.config` exist? If not, try creating it and try `yarn global add serve again`

@Grimler91 I'm sure I have `~/.config` exist.

**Comment by Grimler91** [comment]
Maybe a hard link problem then, that needs to be patched. I can't reproduce the error on my device

**Comment by kalmari246** [comment]
```
Arguments: 
  /data/data/com.termux/files/usr/bin/node /data/data/com.termux/files/usr/share/yarn/bin/yarn.js global add neovim

PATH: 
  /data/data/com.termux/files/usr/opt/rust-nightly/bin:/data/data/com.termux/files/usr/opt/rust-nightly/bin:/data/data/com.termux/files/usr/bin:/data/data/com.termux/files/usr/bin/applets:/data/data/com.termux/files/home/.cargo/bin

Yarn version: 
  1.21.1

Node version: 
  13.0.0

Platform: 
  android arm64

Trace: 
  Error: EACCES: permission denied, copyfile '/data/data/com.termux/files/home/.cache/yarn/v6/npm-@msgpack-msgpack-1.10.0-f120822e641deef03aa0d2cfa2f257becd48d149-integrity/node_modules/@msgpack/msgpack/package.json' -> '/data/data/com.termux/files/home/.config/yarn/global/node_modules/@msgpack/msgpack/package.json'

npm manifest: 
  No manifest

yarn manifest: 
  No manifest

Lockfile: 
  No lockfile
```
same

**Comment by kalmari246** [comment]
```
Packages CPU architecture:
aarch64
Subscribed repositories:
# sources.list
deb https://termux.org/packages/ stable main
# x11-repo (sources.list.d/x11.list)
deb https://dl.bintray.com/xeffyr/x11-packages x11 main
# game-repo (sources.list.d/game.list)
deb https://dl.bintray.com/grimler/game-packages-24 games stable
# unstable-repo (sources.list.d/unstable.list)
deb https://dl.bintray.com/xeffyr/unstable-packages unstable main
# science-repo (sources.list.d/science.list)
deb https://dl.bintray.com/grimler/science-packages-24 science stable
Updatable packages:
busybox/stable 1.31.1-4 aarch64 [upgradable from: 1.31.1-3]
Android version:
10
Kernel build information:
Linux localhost 4.14.113-17369399 #1 SMP PREEMPT Tue Nov 26 23:09:17 KST 2019 aarch64 Android
Device manufacturer:
samsung
Device model:
SM-G975F
```
im on android 10 as well

**Comment by kalmari246** [comment]
```
$ proot -l yarn global add neovim
yarn global v1.21.1
[1/4] Resolving packages...
[2/4] Fetching packages...
[3/4] Linking dependencies...
error An unexpected error occurred: "EACCES: permission denied, copyfile '/data/data/com.termux/files/home/.cache/yarn/v6/npm-color-convert-1.9.3-bb71850690e1f136567de629d2d5471deda4c1e8-integrity/node_modules/color-convert/index.js' -> '/data/data/com.termux/files/home/.config/yarn/global/node_modules/color-convert/index.js'".
info If you think this is a bug, please open a bug report with the information provided in "/data/data/com.termux/files/home/.config/yarn/global/yarn-error.log".
info Visit https://yarnpkg.com/en/docs/cli/global for documentation about this command.
```
`proot --link2symlink` doesnt work either

**Comment by xeffyr** [comment]
Fixed in nodejs v13.0.0-1 (similar patch is applied to nodejs-lts too).
